hq/app/logic/IamRemediation.scala

```diff
@@ -106,7 +106,24 @@ object IamRemediation extends Logging {
     * This might fail, because it may be that no matching key exists.
     */
   def lookupCredentialId(badKeyCreationDate: DateTime, userCredentials: List[CredentialMetadata]): Attempt[CredentialMetadata] = {
-    ???
+    userCredentials.filter { credentialMetadata =>
+      credentialMetadata.creationDate.withTimeAtStartOfDay() == badKeyCreationDate.withTimeAtStartOfDay()
+    } match {
+      case singleMatchingKey :: Nil => Attempt.Right(singleMatchingKey)
+      case Nil => Attempt.Left(FailedAttempt(Failure(
```

Comment:
It might be a little neater to define a separate type for these specific failures and then it might also be easier in the tests to check for a specific failure rather than just `isFailedAttempt`

Reply:
Let's discuss this.